common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -533,6 +533,18 @@ uint32_t network_utils::linux_get_iface_index(const std::string &iface_name)
     return iface_index;
 }
 
+std::string network_utils::linux_get_iface_name(uint32_t iface_index)
+{
+    char iface_name[IF_NAMESIZE] = {0};
+    if (!if_indextoname(iface_index, iface_name)) {
+        LOG(ERROR) << "Failed to read the name of interface with index " << iface_index << ": "
```

Comment:
(nitpick) Although at first sight it makes sense to log an error here, the problem is that when the error occurs, you have absolutely no context from the log - except maybe a hint from the preceding message, but that would typically only be if DEBUG is enabled.
Therefore, I think in general it is better to not do any logging in the utility functions, but instead do it in the caller. Well, to be very precise: only do logging of information that is not propagated to the caller (e.g. if there's an errno to report).

Reply:
This is one of the reasons I prefer using exceptions.